doc/ROADMAP.rst.txt

```diff
@@ -0,0 +1,326 @@
+consistently when we add new code, but for 1.0 it should also be done for
+existing code.
+
+
+Test coverage
+`````````````
+Test coverage of code added in the last few years is quite good, and we aim for
+a high coverage for all new code that is added.  However, there is still a
+significant amount of old code for which coverage is poor.  Bringing that up to
+the current standard is probably not realistic, but we should plug the biggest
+holes.  Additionally the coverage should be tracked over time and we should
+ensure it only goes up.
+
+Besides coverage there is also the issue of correctness - older code may have a
+few tests that provide decent statement coverage, but that doesn't necessarily
+say much about whether the code does what it says on the box.  Therefore code
+review of some parts of the code (``stats`` and ``signal`` in particular) is
+necessary.
+
+
+Documentation
+`````````````
+The documentation is in decent shape.  Expanding of current docstrings and
+putting them in the standard numpy format should continue, so the number of
+reST errors and glitches in the html docs decreases.  Most modules also have a
+tutorial in the reference guide that is a good introduction, however there are
+a few missing or incomplete tutorials - this should be fixed.
+
+
+Other
+`````
+Scipy 1.0 will likely contain more backwards-incompatible changes than a minor
+release.  Therefore we will have a longer-lived maintenance branch of the last
+0.X release.
+
+It's not clear how much functionality can be Cythonized without making the .so
+files too large.  This needs measuring.
+
+Bento will be officially supported as the second build tool besides distutils.
+At the moment it still has an experimental, use-at-your-own-risk status, but
+that has to change. 
+
+A more complete continuous integration setup is needed; at the moment we often
+find out right before a release that there are issues on some less-often used
+platform or Python version.  At least needed are a Windows, Linux and OS X
+build, coverage of the lowest and highest Python and Numpy versions that are
+supported, a Bento build and a PEP8 checker.
+
+
+Modules
+-------
+
+cluster
+```````
+Most of the cluster module is a candidate for a Cython rewrite; this will speed
+up the code and it will be more maintainable than the current C code.  The code
+should remain (or become) simple and easy to understand.  Support for the
+arbitrary distance metrics in ``scipy.spatial`` is probably best left to
+scikit-learn or other more specialized libraries.
+
+
+constants
+`````````
+This module is basically done, low-maintenance and without open issues.
+
+
+fftpack
+```````
+Needed:
+
+  - solve issues with single precision: large errors, disabled for difficult sizes
+  - fix caching bug
+  - Bluestein algorithm nice to have, padding is alternative
+  - deprecate fftpack.convolve as public function (was not meant to be public),
```

Comment:
@rifasaurousrex I was going to propose that the existing `signal.convolve()` have a `mode='circ'` option, but I wasn't sure if there was a practical use for it.